momepy/utils.py

```diff
@@ -53,8 +55,22 @@ def _generate_primal(G, gdf_network, fields, multigraph, oneway_column=None):
     Helper for gdf_to_nx.
     """
     G.graph["approach"] = "primal"
+
+    if not "LineString" in gdf_network.geom_type.unique():
+        warnings.warn(
+            message="The given network does not contain any LineString. This can lead to unexpected behaviour. The intended usage of the conversion function is with networks made of LineStrings only.",
```

Comment:
Maybe a nitpick, but since these are nearly the message we could declare it as a variable:

```python
msg = (
    "The given network %s. This can lead to unexpected behaviour. The intended usage of the conversion function is with networks made of LineStrings only."
)
if not "LineString" in gdf_network.geom_type.unique():
        warnings.warn(
             message=msg % "does not contain any LineString.",
             category=RuntimeWarning,
        )
if len(gdf_network.geom_type.unique()) > 1:
        warnings.warn(
            message=msg % "consists of multiple geometry types.",
             category=RuntimeWarning,
         )

Reply:
I like the idea of putting the shared part of the message into a shared variable. This way doubled code is reduced.
I adjusted the way things are splitted up a little bit -> the message is splitted up into a common and a special part on a sentence level. From my point of view this way the code readability is improved.